move_tree.rb

```diff
@@ -0,0 +1,41 @@
+require_relative 'square'
+
+class MoveTree
+
+  attr_reader :root, :max_depth, :num_of_nodes
+
+  def initialize(coordinates,max_depth)
+    @root = Square.new(coordinates[0],coordinates[1],0,nil,[])
+    @max_depth = max_depth
+    @num_of_nodes = 1
+    attach_children(@root)
+  end
+
+  def inspect
+    puts "This tree has #{num_of_nodes} nodes and a max depth of #{max_depth}."
+  end
+
+  private
+
+  def moves
+    [[-2,-1],[-2,1],[-1,-2],[-1,2],[1,-2],[1,2],[2,-1],[2,1]]
+  end
+
+  def on_the_board?(coordinates)
+    coordinates.all? { |coord| (0..7).include? coord }
+  end
+
+  def attach_children(node)
+    moves.each do |move|
```

Comment:
Oh man, what a handy way to do it with the moves. What I did was a case when for 8 moves iterating through all 8, which would also check if they were a valid move. I think I like this way more.